readthedocs/search/views.py

```diff
@@ -86,10 +77,65 @@ def elastic_search(request):
             ),
         )
 
+    if results:
+        if user_input.type == 'file':
+            # Change results to turn newlines in highlight into periods
+            # https://github.com/rtfd/readthedocs.org/issues/5168
+            for result in results:
+                if hasattr(result.meta.highlight, 'content'):
+                    result.meta.highligh.content = [result.replace(
+                        '\n', '. ') for result in result.meta.highligh.content]
+
+        log.debug('Search results: %s', pformat(results.to_dict()))
+        log.debug('Search facets: %s', pformat(results.facets.to_dict()))
+
     template_vars = user_input._asdict()
     template_vars.update({'results': results, 'facets': facets})
     return render(
         request,
         'search/elastic_search.html',
         template_vars,
     )
+
+
+def elastic_project_search(request, project_slug):
+    """Use elastic search to search in a project."""
+    queryset = Project.objects.protected(request.user)
```

Comment:
Why `.protected` is used here instead of `.public`? If it's because of .com, shouldn't be a combination of `.public` + `.for_user`?

Reply:
This is what all the dashboard views in `projects.views.public` do.